<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>omXplore: a versatile series of Shiny apps to explore 'omics' data • omXplore</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="omXplore: a versatile series of Shiny apps to explore 'omics' data">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">omXplore</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.91</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/omXplore.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/addingThirdPartyPlots.html">Adding third party plots</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/prostarproteomics/omXplore/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>omXplore: a versatile series of Shiny apps to
explore ‘omics’ data</h1>
                        <h4 data-toc-skip class="author">Samuel
Wieczorek</h4>
                                    <h4 data-toc-skip class="author">Thomas
Burger</h4>
                        
            <h4 data-toc-skip class="date">2024-10-18</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/prostarproteomics/omXplore/blob/main/vignettes/omXplore.Rmd" class="external-link"><code>vignettes/omXplore.Rmd</code></a></small>
      <div class="hidden name"><code>omXplore.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>The package omXplore (standing for “Omics Explore”) is a R
      package which provides functions to vizualize omics data
      experiments. It deals with common data formats in Bioconductor
      such as MSnSet, MultiAssayEperiments, QFeatures or even simple
      lists.</p>
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The <code>omXplore</code> package offers a series of built-in plots
dedicated to the visualization and the analysis of *omics (genomic,
transcriptomics, proteomics) data. As for several R packages available
in <code>Bioconductor</code> for exploring omics-like datasets,
<code>omXplore</code> is based on Shiny to make those plots easily
available in a web application. Four popular Bioconductor data objects
are currently supported: <code>SummarizedExperiment</code>,
<code>MultiAssayExperiment</code>, <code>MSnset</code> and
<code>QFeatures.</code>It is also possible to use
<code>data.frame</code> or <code>matrix</code> (or lists of) which
contains quantitative data tables (rows for features and columns for
samples).</p>
<p>All these formats are automatically converted into an internal S4
class which is an enriched version of the
<code>MultiAssayexperiment</code> class. This process is invisible to
the end-user.</p>
<p>The package <code>omXplore</code> was created to be versatile,
reusable and scalable. It differs from similar R packages in two main
points:</p>
<ul>
<li><p>(versatile) The main Shiny module in <code>omXplore</code> is a
hub which gives access to the individual plots. The plots are
automatically updated w.r.t the selected dataset (in case of data which
contains several ExperimentData like the classes
<code>MultiAssayExperiment</code> or <code>QFeatures</code>).</p></li>
<li><p>(scalable) with less effort, it is easy integrate external plots
(written as Shiny modules) in the main GUI of
<code>omXplore</code>.</p></li>
<li><p>(reusable) Each plot (a Shiny module) can be run alone or
integrated as a complementary tool in third party Shiny apps. As an
example, it is well suited for the package <a href="https://www.bioconductor.org/packages/release/bioc/html/Prostar.html" class="external-link">Prostar</a>
in which it is used.</p></li>
</ul>
</div>
<div class="section level2">
<h2 id="features">Features<a class="anchor" aria-label="anchor" href="#features"></a>
</h2>
<p><code>omXplore</code> provides a graphical user interface using the
<code>Shiny</code> and the <code>highcharter</code> packages for the
following visualizations:</p>
<ul>
<li>Connected Components of graph-type data (e.g. in proteomics
datasets, graphs of peptide-protein relationship),</li>
<li>Principal Component Analysis (PCA),</li>
<li>Histograms to analyze the quantitative data based on cell metadata
(eg missing values, imputed values, …),</li>
<li>Intensity plots (boxplot and violinplot) of quantitative data. There
is also a tool to select and view (by superimposition) the evolution of
intensity over samples for a given set of entities,</li>
<li>Density plot</li>
<li>Variance distribution plot</li>
<li>Heatmap</li>
<li>Correlation plot</li>
</ul>
<p>For developers or users who wants to enhance their application,
additional features include:</p>
<ul>
<li>As each plot is a Shiny module, it can be launched in a standalone
mode (from a R console) or it can be easily integrated into any third
party Shiny app</li>
<li>Internal convert function from a large variety of
<code>Bioconductor</code> objects,</li>
<li>A easy way to add user-defined plots (written as Shiny modules) in
the main GUI of <code>omXplore</code>
</li>
</ul>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install this package, start R (version “4.3”) and enter:</p>
<pre><code><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://bioconductor.github.io/BiocManager/" class="external-link">"BiocManager"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"omXplore"</span><span class="op">)</span></span></code></pre>
<p>This will also install dependencies.</p>
<p>It is also possible to install <code>omXplore</code> from Github:</p>
<pre><code><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu">install_github</span><span class="op">(</span><span class="st">'prostarproteomics/omXplore'</span><span class="op">)</span></span></code></pre>
<p>Then, load the package into R environment:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/prostarproteomics/omXplore" class="external-link">"omXplore"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="enriching-native-multiassayexperiment">Enriching native MultiAssayExperiment<a class="anchor" aria-label="anchor" href="#enriching-native-multiassayexperiment"></a>
</h2>
<p>The internal data struture used in <code>omXplore</code> is based on
the class <a href="https://www.bioconductor.org/packages/release/bioc/html/MultiAssayExperiment.html" class="external-link">MultiAssayExperiments</a>
which is enriched with specific slots needed to create built-in plots.
The plots display statistical information about data contained in the
SummarizedExperiment slots of a dataset.</p>
<p>Some items are added to the metadata of the instance of
MultiAssayExperiment and to each of the instances of
SummarizedExperiment.</p>
<p><strong>Additional info in MultiAssayExperiment</strong></p>
<p>Currently, there is no custom slot in the metadata of the mae
contains the following additional items (in the slot names
<code>other</code>):</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">vdata</span><span class="op">)</span></span>
<span><span class="fu">MultiAssayExperiment</span><span class="fu">::</span><span class="fu">metadata</span><span class="op">(</span><span class="va">vdata</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $other</span></span>
<span><span class="co">## list()</span></span></code></pre>
<p><strong>Additional info in SummarizedExperiment</strong></p>
<p>The metadata of each SummarizedExperiment dataset contains the
following items:</p>
<ul>
<li><p><strong>proteinID</strong>: the name of the column which contains
xxx</p></li>
<li><p><strong>colID</strong>: the name of the column which serves as
unique index</p></li>
<li><p><strong>cc</strong>: the list of Connected Components, based on
the adjacency matrix (if exists)</p></li>
<li><p><strong>type</strong>: the type of data contained in the current
Experiment (e.g. peptide, protein, …)</p></li>
<li><p><strong>pkg_version</strong>: the name and version number which
has been used to create the current Experiment.</p></li>
</ul>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">vdata</span><span class="op">)</span></span>
<span><span class="fu">MultiAssayExperiment</span><span class="fu">::</span><span class="fu">metadata</span><span class="op">(</span><span class="va">vdata</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $pkg_version</span></span>
<span><span class="co">## NULL</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $type</span></span>
<span><span class="co">## [1] "protein"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $colID</span></span>
<span><span class="co">## [1] "protID"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $proteinID</span></span>
<span><span class="co">## [1] "protID"</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $cc</span></span>
<span><span class="co">## $cc[[1]]</span></span>
<span><span class="co">## 1 x 1 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##        proteinID_1</span></span>
<span><span class="co">## prot_1           1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $cc[[2]]</span></span>
<span><span class="co">## 1 x 1 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##        proteinID_2</span></span>
<span><span class="co">## prot_2           1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $cc[[3]]</span></span>
<span><span class="co">## 1 x 1 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##        proteinID_3</span></span>
<span><span class="co">## prot_3           1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $cc[[4]]</span></span>
<span><span class="co">## 1 x 1 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##        proteinID_4</span></span>
<span><span class="co">## prot_4           1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $cc[[5]]</span></span>
<span><span class="co">## 1 x 1 sparse Matrix of class "dgCMatrix"</span></span>
<span><span class="co">##        proteinID_5</span></span>
<span><span class="co">## prot_5           1</span></span></code></pre>
<p>The <strong>adjacencyMatrix</strong> (when exists) is stored as a
DataFrame in the rowData() of a SummarizedExperiment itemµ.</p>
<p>All modules are self-contained in the sense that it is not necessary
to manipulate datasets to view the plots. The information described
above are given only to discover the slots used if the user wants to
enrich its dataset before using omXplore.</p>
</div>
<div class="section level2">
<h2 id="using-omxplore">Using omXplore<a class="anchor" aria-label="anchor" href="#using-omxplore"></a>
</h2>
<p>The package <code>omXplore</code> offers a collection of standard
plots written as Shiny modules. The main app is a Shiny module itself
which displays the plots of each module.</p>
<p>This section describes how to view built-in plots and the main app of
<code>omXplore</code>.</p>
<div class="section level3">
<h3 id="individual-built-in-plots">Individual built-in plots<a class="anchor" aria-label="anchor" href="#individual-built-in-plots"></a>
</h3>
<p>The list of plots available in the current R session via omXplore can
be obtained with:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/omXplore-modules.html">listPlotModules</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "omXplore_cc"          "omXplore_corrmatrix"  "omXplore_density"    </span></span>
<span><span class="co">## [4] "omXplore_heatmap"     "omXplore_intensity"   "omXplore_pca"        </span></span>
<span><span class="co">## [7] "omXplore_tabExplorer" "omXplore_variance"</span></span></code></pre>
<p>By default, this function lists the built-in modules and the external
modules compliant with omXplore.</p>
<p>Each of these functions is a Shiny app implemented as a module and
can be launched in a standalone mode or embedded in another shiny app
(as it is the case with the main UI of <code>omXplore</code> or inserted
in a third party Shiny app).</p>
<p>Most of these functions analyse the data contained in an Experiment
of the dataset (an instance of the class
<code>SummarizedExperiment</code>). For a sake of simplicity, they all
have the same two parameters: (1) the dataset in any (compatible) format
(See the help page of the plot functions for details) and (2) the indice
of the assay to analyse (See <a href="https://www.bioconductor.org/packages/release/bioc/html/MultiAssayExperiment.html" class="external-link">MultiAssayExperiment</a>).</p>
<p>Internally, each function builds the enriched instance of MAE used
inside omXplore then show the plot for the assay which has been
specified in parameters.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">sub_R25</span><span class="op">)</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/density-plot.html">omXplore_density</a></span><span class="op">(</span><span class="va">sub_R25</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span><span class="op">)</span></span></code></pre></div>
<p>Note: this code to run a shiny app follows the recommendations of
Bioconductor on <a href="https://contributions.bioconductor.org/shiny.html#running-app" class="external-link">Running
Shiny apps</a>.</p>
<div class="figure" style="text-align: center">
<img src="figs%2FomXplore_density.png" alt="Plot generated by the module omXplore_density()" width="100%"><p class="caption">
Plot generated by the module omXplore_density()
</p>
</div>
</div>
<div class="section level3">
<h3 id="main-ui">Main UI<a class="anchor" aria-label="anchor" href="#main-ui"></a>
</h3>
<p>As it is 9described in the previous section, omXplore have several
built-in plots. And it may be fastidious to launch each plot function
one after one to completely analyze a dataset.</p>
<p>For that purpose, <code>omXplore</code> has another shiny app, called
<code><a href="../reference/ds-view.html">view_dataset()</a></code> which acts as a hub for plots to facilitate
the analyse of the different assays in a dataset. It is launched as
follows:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">sub_R25</span><span class="op">)</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/ds-view.html">view_dataset</a></span><span class="op">(</span><span class="va">sub_R25</span><span class="op">)</span></span>
<span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span><span class="op">)</span></span></code></pre></div>
<p>The resulting UI is the following:</p>
<div class="figure" style="text-align: center">
<img src="figs%2Fexample_view_dataset_nomodal.png" alt="`omXplore` interactive interface with modal." width="100%"><p class="caption">
<code>omXplore</code> interactive interface with modal.
</p>
</div>
<p>The interface is divided in three parts.</p>
<p><strong>(A) Choosing the assay</strong></p>
<p>A widget let the user select one of the experiments contained in the
dataset.</p>
<p><strong>(B) Select which plot to display</strong></p>
<p>A series of clickable vignettes which represent the different plots
available. When the user clicks on a vignette, the corresponding plot is
displayed (See are C).</p>
<p><strong>(C) Viewing the plots</strong></p>
<p>The plots are displayed in the same window as the UI (below the
vignettes) or in a modal window, depending of the option used to launch
the Shinyp app (See <code><a href="../reference/ds-view.html">?view_dataset</a></code>).</p>
<div class="figure" style="text-align: center">
<img src="figs%2Fview_dataset_modal.png" alt="`omXplore` interactive interface with modal." width="100%"><p class="caption">
<code>omXplore</code> interactive interface with modal.
</p>
</div>
<p>When a plot is displayed, it shows the data corresponding to the
dataset selected in the widget (of the left side). If this dataset is
changed, the plot is automatically updated with the data of the new
dataset.</p>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] omXplore_0.99.91 BiocStyle_2.32.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RColorBrewer_1.1-3          jsonlite_1.8.9             </span></span>
<span><span class="co">##   [3] MultiAssayExperiment_1.30.3 magrittr_2.0.3             </span></span>
<span><span class="co">##   [5] shinyjqui_0.4.1             estimability_1.5.1         </span></span>
<span><span class="co">##   [7] MALDIquant_1.22.3           rmarkdown_2.28             </span></span>
<span><span class="co">##   [9] fs_1.6.4                    zlibbioc_1.50.0            </span></span>
<span><span class="co">##  [11] ragg_1.3.3                  vctrs_0.6.5                </span></span>
<span><span class="co">##  [13] htmltools_0.5.8.1           S4Arrays_1.4.1             </span></span>
<span><span class="co">##  [15] curl_5.2.3                  broom_1.0.7                </span></span>
<span><span class="co">##  [17] SparseArray_1.4.8           mzID_1.42.0                </span></span>
<span><span class="co">##  [19] TTR_0.24.4                  sass_0.4.9                 </span></span>
<span><span class="co">##  [21] KernSmooth_2.23-24          bslib_0.8.0                </span></span>
<span><span class="co">##  [23] htmlwidgets_1.6.4           desc_1.4.3                 </span></span>
<span><span class="co">##  [25] plyr_1.8.9                  impute_1.78.0              </span></span>
<span><span class="co">##  [27] emmeans_1.10.5              zoo_1.8-12                 </span></span>
<span><span class="co">##  [29] lubridate_1.9.3             cachem_1.1.0               </span></span>
<span><span class="co">##  [31] igraph_2.0.3                mime_0.12                  </span></span>
<span><span class="co">##  [33] iterators_1.0.14            lifecycle_1.0.4            </span></span>
<span><span class="co">##  [35] pkgconfig_2.0.3             Matrix_1.7-0               </span></span>
<span><span class="co">##  [37] R6_2.5.1                    fastmap_1.2.0              </span></span>
<span><span class="co">##  [39] shiny_1.9.1                 GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">##  [41] MatrixGenerics_1.16.0       clue_0.3-65                </span></span>
<span><span class="co">##  [43] digest_0.6.37               pcaMethods_1.96.0          </span></span>
<span><span class="co">##  [45] colorspace_2.1-1            S4Vectors_0.42.1           </span></span>
<span><span class="co">##  [47] textshaping_0.4.0           GenomicRanges_1.56.2       </span></span>
<span><span class="co">##  [49] fansi_1.0.6                 timechange_0.3.0           </span></span>
<span><span class="co">##  [51] httr_1.4.7                  abind_1.4-8                </span></span>
<span><span class="co">##  [53] compiler_4.4.1              doParallel_1.0.17          </span></span>
<span><span class="co">##  [55] backports_1.5.0             BiocParallel_1.38.0        </span></span>
<span><span class="co">##  [57] viridis_0.6.5               highr_0.11                 </span></span>
<span><span class="co">##  [59] dendextend_1.18.1           gplots_3.2.0               </span></span>
<span><span class="co">##  [61] MASS_7.3-60.2               DelayedArray_0.30.1        </span></span>
<span><span class="co">##  [63] scatterplot3d_0.3-44        gtools_3.9.5               </span></span>
<span><span class="co">##  [65] caTools_1.18.3              mzR_2.38.0                 </span></span>
<span><span class="co">##  [67] flashClust_1.01-2           tools_4.4.1                </span></span>
<span><span class="co">##  [69] PSMatch_1.8.0               httpuv_1.6.15              </span></span>
<span><span class="co">##  [71] quantmod_0.4.26             FactoMineR_2.11            </span></span>
<span><span class="co">##  [73] glue_1.8.0                  promises_1.3.0             </span></span>
<span><span class="co">##  [75] QFeatures_1.14.2            grid_4.4.1                 </span></span>
<span><span class="co">##  [77] reshape2_1.4.4              cluster_2.1.6              </span></span>
<span><span class="co">##  [79] generics_0.1.3              gtable_0.3.5               </span></span>
<span><span class="co">##  [81] preprocessCore_1.66.0       shinyBS_0.61.1             </span></span>
<span><span class="co">##  [83] sm_2.2-6.0                  tidyr_1.3.1                </span></span>
<span><span class="co">##  [85] data.table_1.16.2           utf8_1.2.4                 </span></span>
<span><span class="co">##  [87] XVector_0.44.0              BiocGenerics_0.50.0        </span></span>
<span><span class="co">##  [89] foreach_1.5.2               ggrepel_0.9.6              </span></span>
<span><span class="co">##  [91] pillar_1.9.0                stringr_1.5.1              </span></span>
<span><span class="co">##  [93] limma_3.60.6                later_1.3.2                </span></span>
<span><span class="co">##  [95] dplyr_1.1.4                 lattice_0.22-6             </span></span>
<span><span class="co">##  [97] tidyselect_1.2.1            vioplot_0.5.0              </span></span>
<span><span class="co">##  [99] knitr_1.48                  gridExtra_2.3              </span></span>
<span><span class="co">## [101] bookdown_0.41               IRanges_2.38.1             </span></span>
<span><span class="co">## [103] ProtGenerics_1.36.0         SummarizedExperiment_1.34.0</span></span>
<span><span class="co">## [105] stats4_4.4.1                xfun_0.48                  </span></span>
<span><span class="co">## [107] Biobase_2.64.0              statmod_1.5.0              </span></span>
<span><span class="co">## [109] factoextra_1.0.7            MSnbase_2.30.1             </span></span>
<span><span class="co">## [111] matrixStats_1.4.1           DT_0.33                    </span></span>
<span><span class="co">## [113] visNetwork_2.1.2            stringi_1.8.4              </span></span>
<span><span class="co">## [115] UCSC.utils_1.0.0            lazyeval_0.2.2             </span></span>
<span><span class="co">## [117] yaml_2.3.10                 evaluate_1.0.1             </span></span>
<span><span class="co">## [119] codetools_0.2-20            MsCoreUtils_1.16.1         </span></span>
<span><span class="co">## [121] tibble_3.2.1                BiocManager_1.30.25        </span></span>
<span><span class="co">## [123] affyio_1.74.0               multcompView_0.1-10        </span></span>
<span><span class="co">## [125] cli_3.6.3                   xtable_1.8-4               </span></span>
<span><span class="co">## [127] systemfonts_1.1.0           munsell_0.5.1              </span></span>
<span><span class="co">## [129] jquerylib_0.1.4             Rcpp_1.0.13                </span></span>
<span><span class="co">## [131] GenomeInfoDb_1.40.1         XML_3.99-0.17              </span></span>
<span><span class="co">## [133] parallel_4.4.1              leaps_3.2                  </span></span>
<span><span class="co">## [135] pkgdown_2.1.1               ggplot2_3.5.1              </span></span>
<span><span class="co">## [137] assertthat_0.2.1            AnnotationFilter_1.28.0    </span></span>
<span><span class="co">## [139] bitops_1.0-9                viridisLite_0.4.2          </span></span>
<span><span class="co">## [141] mvtnorm_1.3-1               rlist_0.4.6.2              </span></span>
<span><span class="co">## [143] affy_1.82.0                 scales_1.3.0               </span></span>
<span><span class="co">## [145] xts_0.14.1                  ncdf4_1.23                 </span></span>
<span><span class="co">## [147] purrr_1.0.2                 highcharter_0.9.4          </span></span>
<span><span class="co">## [149] crayon_1.5.3                rlang_1.1.4                </span></span>
<span><span class="co">## [151] vsn_3.72.0                  shinyjs_2.1.0</span></span></code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Samuel Wieczorek, Thomas Burger.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
